src/app/core/services/show-style-variant-state.service.ts

```diff
@@ -3,13 +3,40 @@ import { BehaviorSubject, lastValueFrom, Observable } from 'rxjs'
 import { ShowStyleVariant } from '../models/show-style-variant'
 import { ShowStyleVariantService } from '../abstractions/show-style-variant.service'
 import { EventSubscription } from '../../event-system/abstractions/event-observer.service'
+import { RundownEventObserver } from './rundown-event-observer.service'
+import { RundownUpdatedEvent } from '../models/rundown-event'
+import {Logger} from "../abstractions/logger.service";
 
 @Injectable()
 export class ShowStyleVariantStateService implements OnDestroy {
   private readonly showStyleVariantSubjects: Map<string, BehaviorSubject<ShowStyleVariant>> = new Map()
   private readonly eventSubscriptions: EventSubscription[]
+  private readonly logger: Logger
 
-  constructor(private readonly showStyleVariantService: ShowStyleVariantService) {}
+  constructor(
+    private readonly showStyleVariantService: ShowStyleVariantService,
+    private readonly rundownEventObserver: RundownEventObserver,
+    logger: Logger
+  ) {
+    this.logger = logger.tag('ShowStyleVariantStateService')
+    this.subscribeToRundownEvents()
+  }
+
+  private subscribeToRundownEvents(): void {
```

Comment:
It seems a little overkill to make entire method that says it subscibes to multiple RundownEvents and then it only subscribes to one.

With the current state of this file, it would be fine to have the one-liner, that is the body of this method, in the constructor.

Reply:
It has been removed and now happens in the constructor!